docs/custom-metric-filter.md

```diff
@@ -0,0 +1,151 @@
+Cloud Monarch backend
+- `prometheus` stands for the configuration that exports metrics in Prometheus
+format and later can be scraped by a prometheus server following [this guide](http://cloud/anthos-config-management/docs/how-to/monitoring-config-sync#prometheus)
+
+Here is [the current configuration](https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/pkg/metrics/otel.go#L38) of Open Telemetry Collector
+in Config Sync.
+
+## Patch the otel collector deployment with ConfigMap
+
+1. **Get the current ConfigMap as template**
+
+    ```
+    kubectl get cm otel-collector-googlecloud -n config-management-monitoring -o yaml > otel-collector-config.yaml
+    ```
+    
+    Or in Config Sync with version prior to `v1.11.1`, use
+    
+    ```
+    kubectl get cm otel-collector-stackdriver -n config-management-monitoring -o yaml > otel-collector-config.yaml
+    ```
+1. Change the `name` under `.data.otel-collector-config.yaml` to `otel-collector-custom`
+
+    This step spawns a new ConfigMap with name `otel-collector-custom` that is recognized by Config Sync, which overrides the `otel-collector-googlecloud` ConfigMap. Documents of the custom collector can be found [here](http://cloud/anthos-config-management/docs/how-to/monitoring-config-sync#custom-exporter). 
+
+1. **Modify the otel-collector-config.yaml file**
+
+    - To **add or drop metrics** from the
+       [available metrics](http://cloud/anthos-config-management/docs/how-to/monitoring-config-sync#metrics),
+       modify the `include` section under the
+       [filter processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/filterprocessor)
+       configuration
+
+       > Note: If running Config Sycn version prior to 1.11.x, use the
+       [default config](https://github.com/tiffanny29631/kpt-config-sync/blob/main/pkg/metrics/otel.go#L38)
+
+       ```
+       # This is an example adding crd_count metric
+       # Make sure you are modifying the `filter/cloudmonitoring`
+       ...
+       processors:
+         ...
+         filter/cloudmonitoring:
+           metrics:
+             include:
+               match_type: regexp
+               metric_names:
+                 - reconciler_errors
+                 - pipeline_error_observed
+                 - declared_resources
+                 - apply_operations_total
+                 - resource_fights_total
+                 - internal_errors_total
+                 - kcc_resource_count
+                 - resource_count
+                 - ready_resource_count
+                 - cluster_scoped_resource_count
+                 - resource_ns_count
+                 - api_duration_seconds
+                 - crd_count         <----- Here
+       ```
+
+    - To **turn off the custom metrics** in Cloud Monitoring, remove this section
+
+        Note: In Config Sync versions prior to 1.12.0, the exporter name was `stackdriver`
+        instead of `googlecloud`.
```

Comment:
Note: The `stackdriver` exporter was renamed to `googlecloud` after otel-collector was upgraded in Config Sync 1.12.0.

Reply:
Done